src/OrleansAzureUtils/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -0,0 +1,235 @@
+            settings = new JsonSerializerSettings
+            {
+                TypeNameHandling = TypeNameHandling.Auto,
+                TypeNameAssemblyFormat = FormatterAssemblyStyle.Simple,
+                Formatting = Formatting.None
+            };
+
+            if (config.Properties.ContainsKey("SerializeTypeNames"))
+            {
+                bool serializeTypeNames = false;
+                var serializeTypeNamesValue = config.Properties["SerializeTypeNames"];
+                bool.TryParse(serializeTypeNamesValue, out serializeTypeNames);
+                if (serializeTypeNames)
+                {
+                    settings.TypeNameHandling = TypeNameHandling.All;
+                }
+            }
+
+            if (config.Properties.ContainsKey("PreserveReferencesHandling"))
+            {
+                bool preserveReferencesHandling;
+                var preserveReferencesHandlingValue = config.Properties["PreserveReferencesHandling"];
+                bool.TryParse(preserveReferencesHandlingValue, out preserveReferencesHandling);
+                if (preserveReferencesHandling)
+                {
+                    settings.PreserveReferencesHandling = PreserveReferencesHandling.Objects;
+                }
+            }
+
+            if (config.Properties.ContainsKey("UseFullAssemblyNames"))
+            {
+                bool useFullAssemblyNames = false;
+                var UseFullAssemblyNamesValue = config.Properties["UseFullAssemblyNames"];
+                bool.TryParse(UseFullAssemblyNamesValue, out useFullAssemblyNames);
+                if (useFullAssemblyNames)
+                {
+                    settings.TypeNameAssemblyFormat = FormatterAssemblyStyle.Full;
+                }
+            }
+
+            if (config.Properties.ContainsKey("IndentJSON"))
+            {
+                bool indentJSON = false;
+                var indentJSONValue = config.Properties["IndentJSON"];
+                bool.TryParse(indentJSONValue, out indentJSON);
+                if (indentJSON)
+                {
+                    settings.Formatting = Formatting.Indented;
+                }
+            }
+        }
+
+        /// <summary> Shutdown this storage provider. </summary>
+        /// <see cref="IProvider.Close"/>
+        public Task Close()
+        {
+            return TaskDone.Done;
+        }
+
+        /// <summary> Read state data function for this storage provider. </summary>
+        /// <see cref="IStorageProvider.ReadStateAsync"/>
+        public async Task ReadStateAsync(string grainType, GrainReference grainId, IGrainState grainState)
+        {
+            try
+            {
+                var blobName = GetBlobName(grainType, grainId);
+                var blob = container.GetBlockBlobReference(blobName);
+
+                string text;
+
+                try
+                {
+                    text = await blob.DownloadTextAsync();
```

Comment:
In theory all `await`ed calls in this class should use `.ConfigureAwait(false)`, since non of them require the Orleans context nor scheduler.
Having said that, I think we need to do a thorough review of our current codebase and fix this same thing in other places too, so I wouldn't mind if you don't address it for this particular class in this PR.